mirdata/datasets/cante100.py

```diff
@@ -320,92 +322,82 @@ def to_jams(self):
         )
 
 
-def load_spectrogram(spectrogram_path):
+@io.coerce_to_string_io
+def load_spectrogram(faudio: TextIO) -> np.ndarray:
     """Load a cante100 dataset spectrogram file.
 
     Args:
-        spectrogram_path (str): path to audio file
+        faudio (str or file-like): path or file-like object pointing to an audio file
 
     Returns:
         np.ndarray: spectrogram
 
     """
-    if not os.path.exists(spectrogram_path):
-        raise IOError("spectrogram_path {} does not exist".format(spectrogram_path))
-    parsed_spectrogram = np.genfromtxt(spectrogram_path, delimiter=" ")
+    parsed_spectrogram = np.genfromtxt(faudio, delimiter=" ")
     spectrogram = parsed_spectrogram.astype(np.float)
 
     return spectrogram
 
 
-def load_audio(audio_path):
+def load_audio(faudio: str) -> Tuple[np.ndarray, float]:
     """Load a cante100 audio file.
 
     Args:
-        audio_path (str): path to audio file
+        faudio (str): path to an audio file
 
     Returns:
-        * np.ndarray - the mono audio signal
-        * float - The sample rate of the audio file
+        np.ndarray: the mono audio signal
+        float: The sample rate of the audio file
 
     """
-    if not os.path.exists(audio_path):
-        raise IOError("audio_path {} does not exist".format(audio_path))
-    audio, sr = librosa.load(audio_path, sr=22050, mono=False)
-    return audio, sr
+    return librosa.load(faudio, sr=22050, mono=False)
 
 
-def load_melody(f0_path):
+@io.coerce_to_string_io
+def load_melody(f0: TextIO) -> Optional[annotations.F0Data]:
     """Load cante100 f0 annotations
 
     Args:
-        f0_path (str): path to audio file
+        f0 (str or file-like): path or file-like object pointing to an audio file
 
     Returns:
         F0Data: predominant melody
 
     """
-    if not os.path.exists(f0_path):
-        raise IOError("f0_path {} does not exist".format(f0_path))
-
     times = []
     freqs = []
-    with open(f0_path, "r") as fhandle:
-        reader = csv.reader(fhandle, delimiter=",")
-        for line in reader:
-            times.append(float(line[0]))
-            freqs.append(float(line[1]))
+    reader = csv.reader(f0, delimiter=",")
+    for line in reader:
+        times.append(float(line[0]))
+        freqs.append(float(line[1]))
 
     times = np.array(times)
     freqs = np.array(freqs)
-    confidence = (freqs > 0).astype(float)
+    confidence = (cast(np.ndarray, freqs) > 0).astype(float)
```

Comment:
It's for typing. Because freqs is a List and then converted to an np.array, mypy still thinks it is its original typing (List). `cast`does nothing but help the type checker understand the type has changed. The other option is to assign np.array(freqs) to a different variable.